python/cudf/cudf/core/column/decimal.py

```diff
@@ -0,0 +1,63 @@
+import cupy as cp
+import numpy as np
+import pyarrow as pa
+
+from cudf import _lib as libcudf
+from cudf.core.buffer import Buffer
+from cudf.core.column import ColumnBase
+from cudf.core.dtypes import DecimalDtype
+from cudf.utils.utils import pa_mask_buffer_to_mask
+
+
+class DecimalColumn(ColumnBase):
+    @classmethod
+    def from_arrow(cls, data: pa.Array):
+        mask_buf = data.buffers()[0]
+        mask = (
+            mask_buf
+            if mask_buf is None
+            else pa_mask_buffer_to_mask(mask_buf, len(data))
+        )
+        data_128 = cp.array(np.frombuffer(data.buffers()[1]).view("int64"))
+        data_64 = data_128[::2].copy()
+        return cls(
+            data=Buffer(data_64.view("uint8")),
+            size=len(data),
+            dtype=DecimalDtype.from_arrow(data.type),
+            mask=mask,
+        )
+
+    def to_arrow(self):
+        data_buf_64 = self.base_data.to_host_array().view("int64")
+        zeros_buf = np.zeros(len(data_buf_64), dtype="int64")
+        data_buf_128 = pa.py_buffer(
+            np.ravel((data_buf_64, zeros_buf), order="F")
+        )
```

Comment:
Would it be more efficient to allocate a buffer of zeros of the correct length and then do a strided copy from `data_buf_64` into that buffer? Something like:
```
data_buf_128 = np.zeros(len(data_buf_64)*2, dtype='int64')
data_buf_128[::2] = data_buf_64
```

Reply:
Yup -- I think we were reading [this graph](https://stackoverflow.com/a/48489150) wrong, @codereport.  Fixed to use Keith/"Paul" approach.